packages/design-system/src/components/TextField/useLabelMask.tsx

```diff
@@ -93,37 +93,58 @@ export const SSN_MASK: MaskFunction = makeMask(RE_SSN, '###-##-####', (match) =>
  * Currency mask is a little different, we need to modify the incoming content to strip
  * out any commas or dollar signs before evaluating it via the Intl.NumberFormat function.
  */
-export const CURRENCY_MASK = (rawInput = '', valueOnly = false) => {
-  const signed = rawInput.includes('-');
-  const stripped = rawInput.replace(/[^0-9.]/g, '');
+export const CURRENCY_MASK = makeMask(RE_CURRENCY, '$', (match) => {
+  const signed = match[0].includes('-');
+  const stripped = match[0].replace(/[^0-9.]/g, '');
   const clipped = stripped.includes('.') ? stripped.slice(0, stripped.indexOf('.') + 3) : stripped;
   const USDollar = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' });
-  const formatted = USDollar.format(Number(clipped));
+  const formatted = USDollar.format(Number(clipped)).replace(/\.00/, '');
 
-  return signed ? '-' + formatted : formatted;
-};
+  console.log(Number(clipped));
+
+  if (Number(clipped) > 0) {
+    return signed ? '-' + formatted : formatted;
+  } else {
+    return '';
+  }
+});
 
 export function useLabelMask(maskFn: MaskFunction, originalInputProps: TextInputProps) {
   // TODO: Once we're on React 18, we can use the `useId` hook
   const labelMaskId = useRef(uniqueId('labelmask_')).current;
   const [focused, setFocused] = useState(false);
   const { onFocus, onBlur, onChange } = originalInputProps;
   const value = originalInputProps.value?.toString() ?? '';
+  const [currentValue, setCurrentValue] = useState(value);
+
+  useEffect(() => {
+    setCurrentValue(value);
+  }, [value, setCurrentValue]);
```

Comment:
I haven't been able to break it! This really seems to work. I'm still really puzzled why we get the cursor jump in the controlled version. When I put a console log in here like

```ts
console.log(`value: ${value}, currentValue: ${currentValue}`)
```

these values are always the same in the controlled component story, which is what I would expect. If the value isn't actually changing, then `setCurrentValue` should be a no-op, which shouldn't cause a re-render, which shouldn't cause a jump in the cursor. I guess I should also try to figure out if it's re-rendering anyway.

Reply:
Because this is happening in a hook, I guess it could be more complicated than that and we need to look at the actual components' render cycles. There's a `TextField` rendering a `TextInput` inside a `LabelMask` with a `useLabelMask` hook. This is a task for another day!

Reply:
Phew, yeah that is a complex one but I'll be happy when I understand why it's happening and how to best solve it 💪🏻